libkbfs/backpressure_disk_limiter.go

```diff
@@ -469,6 +480,52 @@ func (bdl *backpressureDiskLimiter) onBlocksDelete(
 	bdl.fileTracker.onBlocksDelete(blockFiles)
 }
 
+func (bdl *backpressureDiskLimiter) onDiskBlockCacheDelete(
+	ctx context.Context, blockBytes int64) {
+	if blockBytes == 0 {
+		return
+	}
+	bdl.lock.Lock()
+	defer bdl.lock.Unlock()
+	bdl.diskCacheByteTracker.onBlocksDelete(blockBytes)
+}
+
+func (bdl *backpressureDiskLimiter) beforeDiskBlockCachePut(
+	ctx context.Context, blockBytes int64) (
+	availableBytes int64, err error) {
+	if blockBytes == 0 {
+		// Better to return an error than to panic in ForceAcquire.
+		return 0, errors.New("backpressureDiskLimiter.beforeDiskBlockCachePut" +
+			" called with 0 blockBytes")
+	}
+	bdl.lock.Lock()
+	defer bdl.lock.Unlock()
+	freeBytes, _, err := bdl.freeBytesAndFilesFn()
+	if err != nil {
+		return 0, err
+	}
+
+	bt := bdl.diskCacheByteTracker
+	bt.updateFree(freeBytes)
+
+	defer func() {
+		if err != nil || availableBytes < 0 {
```

Comment:
All right so I'll move it into the tracker, but then we still have the issue that if `availableBytes > 0`, we'll need to call `afterDiskBlockCachePut`, but otherwise we won't. So we call `afterDiskBlockCachePut` if and only if we have tried to `Put`. Does that sound right?